research/conditional/utils/model_utils.py

```diff
@@ -319,7 +327,7 @@ def get_common_mot_kwargs(args):
     }
 
 
-def get_ff_layer(args):
+def get_ff_layer(args, rank=None):
```

Comment:
is the rank needed to be here? I am pretty sure that the decision of the kind of layer can be made irrespective of the target machine. Should this logic not be moved upward in terms of abstraction levels?

Reply:
I see a possibility of extending LoggingLayer or some other abstraction with FSDP wrapping functionality. Let's discuss it 1 on 1

Reply:
Unfortunately no, because in MoE we need to pass target machine to the constructor. The reason is that we need to initialize FSDP for `ExpertGating` in the very beginning, before we initialize any root modules. We can discuss this offline

Reply:
As discussed, for now this is the best way forward, as there is some nontrivial handling of precision in different operations withing the Module's `forward`